src/components/ProvideLiquidity.tsx

```diff
@@ -0,0 +1,145 @@
+		return (
+			<div className="max-w-lg mx-auto p-6 bg-white rounded-xl shadow-lg">
+				<h2 className="text-3xl font-bold text-gray-800 mb-4">
+					Provide Liquidity
+				</h2>
+				<p className="text-gray-700">Please connect your wallet to continue.</p>
+			</div>
+		);
+	}
+
+	if (isRoutersLoading) {
+		return (
+			<div className="max-w-lg mx-auto p-6 bg-white rounded-xl shadow-lg">
+				<h2 className="text-3xl font-bold text-gray-800 mb-4">
+					Provide Liquidity
+				</h2>
+				<p className="text-gray-700">Loading routers...</p>
+			</div>
+		);
+	}
+
+	if (routersError) {
+		return (
+			<div className="max-w-lg mx-auto p-6 bg-white rounded-xl shadow-lg">
+				<h2 className="text-3xl font-bold text-gray-800 mb-4">
+					Provide Liquidity
+				</h2>
+				<p className="text-red-600">
+					Error fetching routers: {String(routersError)}
+				</p>
+			</div>
+		);
+	}
+
+	if (!matchedRouter) {
+		return (
+			<div className="max-w-lg mx-auto p-6 bg-white rounded-xl shadow-lg">
+				<h2 className="text-3xl font-bold text-gray-800 mb-4">
+					Provide Liquidity
+				</h2>
+				<p className="text-gray-700">
+					No suitable router found (looking for a router with v2.2 and pool
+					creation enabled).
+				</p>
+			</div>
+		);
+	}
+
+	return (
+		<div className="max-w-lg mx-auto p-6 bg-white rounded-xl shadow-lg">
+			<h2 className="text-3xl font-bold text-gray-800 mb-4">
+				Provide Liquidity
+			</h2>
+
+			{/* First Token Input */}
+			<div className="mb-4">
+				<LiquidityInput
+					assets={allAssets}
+					selectedAsset={tokenA}
+					amount={amountA}
+					onAssetChange={handleTokenAChange}
+					onAmountChange={handleAmountAChange}
+					label="Token A"
+				/>
+			</div>
+
+			{/* Second Token Input */}
+			<div className="mb-4">
+				<LiquidityInput
+					assets={allAssets}
+					selectedAsset={tokenB}
+					amount={amountB}
+					onAssetChange={handleTokenBChange}
+					onAmountChange={handleAmountBChange}
+					label="Token B"
+				/>
+			</div>
+
+			{/* Liquidity Summary */}
+			<div className="mb-4">
+				<LiquiditySummary
+					loading={loadingSimulation}
+					error={simulationError}
+					simulation={lpSimulation ?? null}
+					tokenA={tokenA || undefined}
+					tokenB={tokenB || undefined}
+				/>
+			</div>
+
+			{/* Error Display */}
+			{error && (
+				<div className="mb-4 p-3 bg-red-100 text-red-700 rounded">{error}</div>
+			)}
+
+			{/* Provide Button */}
+			<button
+				onClick={handleProvide}
+				disabled={loadingTx || !tokenA || !tokenB || !amountA || !amountB}
+				className={`w-full py-3 px-4 rounded-lg font-medium text-white ${
+					loadingTx || !tokenA || !tokenB || !amountA || !amountB
+						? "bg-gray-400 cursor-not-allowed"
+						: "bg-blue-600 hover:bg-blue-700"
+				}`}
+			>
+				{loadingTx ? "Providing Liquidity..." : "Provide Liquidity"}
+			</button>
+
+			{/* Router Info (for debugging) */}
+			<div className="mt-4 text-sm text-gray-500">
+				<p>Router: {matchedRouter.address}</p>
```

Comment:
Seeing this makes me think that the `matchedRouter` logic is a little bit off: you could not define a router before selecting the 2 assets and the router to create a pool on should be used from the simulation with a type "initial", not just one of our latest routers


![Screenshot 2025-03-07 at 10 58 18](https://github.com/user-attachments/assets/6a2455da-0874-4512-a1ee-0fe360fd766d)